src/Context/products-context.js

```diff
@@ -0,0 +1,31 @@
+import { createContext, useContext, useReducer, useState } from "react";
+// import { productData } from "../data";
+import { filterReducerFtn } from "../Reducer/filterReducer";
+
+const ProductsContext = createContext();
+
+const  ProductsProvider= ({ children }) => {
+
+    const [products , setProducts ] = useState([])
+    const [filterState , dispatchFilterState] = useReducer(filterReducerFtn , {
+        sortBy : "" ,
```

Comment:
You could create a separate object initialState and then just include it here. This would make the code more readable.

Reply:
Thanks again. Created a util for initial filter state